### PR TITLE
True Strength Index indicator

### LINKED ISSUE #4055: True Strength Index (TSI)
Can we add a new TA indicator: True Strength Index (TSI)

https://www.investopedia.com/terms/t/tsi.asp

#### Potential Solution
Implement TSI, including unit tests with third party data.

### PR DESCRIPTION
**Pull request by jhonabreul**
#### Description
Adds the True Strength Index (TSI) indicator, as proposed by issue #4055 

The TSI is a momentum oscillator based on a double smoothing of price changes, as stated [here](https://school.stockcharts.com/doku.php?id=technical_indicators:true_strength_index).

#### Related Issue
Closes #4055 

#### Motivation and Context
This change adds support for the TSI indicator, which previously was not available and solves the issue linked above.

#### Requires Documentation Change
Yes, documentation will need to be included on how to use the indicator as well as the process for calculating it.

#### How Has This Been Tested?
Downloaded [this spreadshit ](https://school.stockcharts.com/doku.php?id=technical_indicators:true_strength_index#:~:text=Click%20here%20to%20download%20this%20spreadsheet%20example)from StockCharts and added the signal column. Tested agains the colums for the indicator (TSI_25_13) and the signal line (Signal_7) 

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which improves implementation)
- [ ] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:
- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

### CODE REVIEW

File: Indicators/TrueStrengthIndex.cs [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+ * limitations under the License.
+*/
+
+using System;
+namespace QuantConnect.Indicators
+{
+    /// <summary>
+    /// This indicator computes the True Strength Index (TSI).
+    /// The True Strength Index is calculated as explained here:
+    /// https://school.stockcharts.com/doku.php?id=technical_indicators:true_strength_index
```
Comment (Martin-Molinero) [inline Indicators/TrueStrengthIndex.cs:22]: Would be nice to add a small summary of how the indicator is calculated 👍

**Review by Martin-Molinero — CHANGES_REQUESTED** [review]
Awesome! Leaving a few requests 👍

File: Indicators/TrueStrengthIndex.cs [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+                return 0m;
+            }
+
+            decimal pc = input.Price - _prevClose;
+            _prevClose = input.Price;
+
+            _pcEma.Update(input.Time, pc);
+            _apcEma.Update(input.Time, Math.Abs(pc));
+
+            if (_pcEma.IsReady && _apcEma.IsReady) {
```
[the remark this answers is not in the document]
Comment (jhonabreul) [reply Indicators/TrueStrengthIndex.cs:103]: I didn't know about extensions, that's nice, thanks!

**Review by Martin-Molinero — APPROVED** [review]
Nice! So clean using indicator extensions 🙌

**Pull request merged**